rex/joint_pd/wind_rose.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+"""
+Wind Rose (wspd - wdir JPD) calculator
+"""
+from concurrent.futures import as_completed
+import gc
+import logging
+import numpy as np
+import os
+import pandas as pd
+
+from rex.joint_pd.joint_pd import JointPD
+from rex.renewable_resource import WindResource
+from rex.utilities.execution import SpawnProcessPool
+from rex.utilities.loggers import log_mem
+
+logger = logging.getLogger(__name__)
+
+
+class WindRose(JointPD):
```

Comment:
I'm kind of torn on whether this should be a subclass or just use jointpd as a set of utilities. Is the WindRose really a concrete version of jointpd? I guess so? Jointpd just seems like more of a utility set. I dont have strong feelings. Just musings...

Reply:
I already have multiple people using/stocked about the WindRose class so I'd like to keep it. The JointPD class was added because the wave guys need to compute Joint Prob Dist for two wave variables.

Reply:
Oh im not suggesting to remove it, just whether it should be a subclass or call JointPD.classmethods()

Reply:
Ah, I think the extra code is worth being able to just give a hub-height

Reply:
Ah I see, thats a good point... Would be a lot cleaner for sure! Will do

Reply:
Just a suggestion. I'm just not sure WindRose is really a version of JointPD ya know? JointPD seems like a utility calculation set and WindRose relies on it heavily.

Reply:
How does this look:
https://github.com/NREL/rex/blob/f2a928043ff5ad74cb3cb39bef9c024d55c7d1eb/rex/joint_pd/joint_pd.py#L416-L473

Reply:
Yeah that works! is that less code?

Reply:
a lot less!